src/cmd/ls.rs

```diff
@@ -373,6 +373,10 @@ pub fn run_cmd<W: Write>(mut w: &mut W, cwd: &Path, cmd: &Cmd, project: &Project
                 if !ui::show_artifact(&name, art, &pat_case, &cmd.search_settings) {
                     continue;
                 }
+                if !match_made && !fmt_set.long && cmd.ty == OutType::List {
```

Comment:
wow, look at that... I put the filter in the middle of the loop
```
if !ui::show_artifact(&name, art, &pat_case, &cmd.search_settings)
```
That was REALLY bad design. 

Please rebase, I just cleaned up the overall design. And sorry that this was harder than I intended!